sklearn/neighbors/quad_tree.pxd

```diff
@@ -0,0 +1,88 @@
+# cython: boundscheck=False
```

Comment:
The plan is to add a `query` method and to use it for neighbors search for 2 and 3 dimensional datasets.
We put this code here in prevision to this usage but made the `_QuadTree` object private to indicate it should not be used outside our private API and it is subject to change.